contracts/Aavegotchi/facets/ItemsRolesRegistryFacet.sol

```diff
@@ -0,0 +1,253 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.1;
+
+import {IERC1155} from "../../shared/interfaces/IERC1155.sol";
+import {IERC165} from "../../shared/interfaces/IERC165.sol";
+import {IERC7589} from "../../shared/interfaces/IERC7589.sol";
+
+import {LibItems} from "../libraries/LibItems.sol";
+import {LibMeta} from "../../shared/libraries/LibMeta.sol";
+import {LibERC1155Marketplace} from "../libraries/LibERC1155Marketplace.sol";
+
+import {Modifiers, ItemType, EQUIPPED_WEARABLE_SLOTS, GotchiEquippedCommitmentsInfo, Aavegotchi, ItemRolesInfo} from "../libraries/LibAppStorage.sol";
+import {IERC1155Receiver} from "@openzeppelin/contracts/token/ERC1155/IERC1155Receiver.sol";
+import {ERC1155Holder, ERC1155Receiver} from "@openzeppelin/contracts/token/ERC1155/utils/ERC1155Holder.sol";
+import {IEventHandlerFacet} from "../WearableDiamond/interfaces/IEventHandlerFacet.sol";
+import {LibERC1155} from "../../shared/libraries/LibERC1155.sol";
+import {EnumerableSet} from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+
+contract ItemsRolesRegistryFacet is Modifiers, IERC7589, ERC1155Holder {
+    using EnumerableSet for EnumerableSet.UintSet;
+    bytes32 public constant UNIQUE_ROLE = keccak256("Player()");
+
+    uint256 public constant MAX_EXPIRATION_DATE = 90 days;
+
+    /** Modifiers **/
+
+    modifier onlyWearables(address _tokenAddress, uint256 _tokenId) {
+        require(_tokenAddress == s.wearableDiamond, "ItemsRolesRegistryFacet: Only Item NFTs are supported");
+        require(
+            s.itemTypes[_tokenId].category == LibItems.ITEM_CATEGORY_WEARABLE,
+            "ItemsRolesRegistryFacet: Only Items of type Wearable are supported"
+        );
+        _;
+    }
+
+    modifier validGrantRoleData(uint64 _expirationDate, bytes32 _role) {
+        require(
+            _expirationDate > block.timestamp && _expirationDate <= block.timestamp + MAX_EXPIRATION_DATE,
+            "ItemsRolesRegistryFacet: invalid expiration date"
+        );
+        require(_role == UNIQUE_ROLE, "ItemsRolesRegistryFacet: role not supported");
+        _;
+    }
+
+    modifier onlyOwnerOrApproved(address _account, address _tokenAddress) {
+        address _sender = LibMeta.msgSender();
+        require(
+            _account == _sender || isRoleApprovedForAll(_tokenAddress, _account, _sender),
+            "ItemsRolesRegistryFacet: account not approved"
+        );
+        _;
+    }
+
+    modifier sameGrantee(
+        uint256 _commitmentId,
+        bytes32 _role,
+        address _grantee
+    ) {
+        require(_grantee != address(0) && _grantee == s.itemRolesCommitmentInfo[_commitmentId].roleAssignment.grantee, "ItemsRolesRegistryFacet: grantee mismatch");
+        _;
+    }
+
+    /** External Functions **/
+
+    function commitTokens(
```

Comment:
please add in natspec to explain the purpose of these functions @ernanirst

Reply:
Fixed!